library/src/main/java/pl/netigen/rodo/GDPRDialogFragment.kt

```diff
@@ -0,0 +1,209 @@
+import pl.netigen.netigenapi.R
+
+class GDPRDialogFragment : AppCompatDialogFragment() {
+
+    var isNoAdsAvailable = false
+
+    val HTTPS_WWW_NETIGEN_PL_PRIVACY_ONLY_FOR_MOBILE_APPS_NAME_APP = "https://www.netigen.pl/privacy/only-for-mobile-apps-name?app="
+    val HTTPS_WWW_NETIGEN_PL_PRIVACY_ONLY_FOR_MOBILE_APPS = "https://www.netigen.pl/privacy/only-for-mobile-apps"
+
+    private var callback: ClickListener? = null
+    private var admobText: Boolean = false
+
+    override fun onStart() {
+        super.onStart()
+        setDialogSize(0.9, 0.9)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        if (dialog != null) {
+            val window = dialog.window
+            if (window != null) {
+                window.requestFeature(Window.FEATURE_NO_TITLE)
+                dialog.window.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+            }
+        }
+        return inflater.inflate(R.layout.dialog_fragment_gdpr, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        if (activity == null) {
+            dismiss()
+            return
+        }
+        setIcon()
+
+        appNameTextViewGdpr.text = getApplicationName(activity!!)
+        setButtons()
+        addTextAdmob()
+    }
+
+    private fun setButtons() {
+        buttonYes.setOnClickListener { v -> callback?.clickYes() }
+        buttonNo.setOnClickListener { v ->
+            callback!!.clickNo()
+            addPolityText()
+        }
+        buttonBack.setOnClickListener { v ->
+            showAdmobText()
+        }
+        if (isNoAdsAvailable) {
+            buttonPay.setOnClickListener { v -> callback?.clickPay() }
+        } else {
+            buttonPay.visibility = View.GONE
+        }
+        buttonPolicy.setOnClickListener { v -> callback?.clickAcceptPolicy() }
+    }
+
+    private fun setIcon() {
+        try {
+            val icon = activity?.packageManager?.getApplicationIcon(activity!!.packageName)
+            appIconImageViewGdpr.setImageDrawable(icon)
+        } catch (e: PackageManager.NameNotFoundException) {
+            e.printStackTrace()
+        }
+    }
+
+    fun getApplicationName(context: Context): String {
+        val applicationInfo = context.applicationInfo
+        val stringId = applicationInfo.labelRes
+        return if (stringId == 0) applicationInfo.nonLocalizedLabel.toString() else context.getString(stringId)
+    }
+
+    fun setIsPayOptions(isNoAdsAvailable: Boolean) {
+        this.isNoAdsAvailable = isNoAdsAvailable
+    }
+
+    fun isNetworkOn(): Boolean {
+        val context = context
+        val cm: ConnectivityManager?
+        if (context == null) {
+            return false
+        }
+        cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
```

Comment:
cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager - here can be thrown TypeCastException

should be:
val connectivityManager = getSystemService(Context.CONNECTIVITY_SERVICE)
      if (connectivityManager is ConnectivityManager) {

Reply:
Change done it now is

```Kotlin
val connectivityManager = context?.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager?
                ?: return false
```

P.S. Try using MARKDOWN so comments with code will be more readable